src/db_adapters/execution_outcomes.rs

```diff
@@ -28,11 +30,17 @@ pub async fn store_execution_outcomes_for_chunk(
     execution_outcomes: &[near_indexer::IndexerExecutionOutcomeWithReceipt],
     shard_id: near_indexer::near_primitives::types::ShardId,
     block_timestamp: u64,
+    receipts_cache: crate::ReceiptsCache,
 ) -> anyhow::Result<()> {
+    let mut receipts_cache_lock = receipts_cache.lock().await;
     let mut outcome_models: Vec<models::execution_outcomes::ExecutionOutcome> = vec![];
     let mut outcome_receipt_models: Vec<models::execution_outcomes::ExecutionOutcomeReceipt> =
         vec![];
     for (index_in_chunk, outcome) in execution_outcomes.iter().enumerate() {
+        // Trying to take the parent Transaction hash for the Receipt from ReceiptsCache
+        let parent_transaction_hash =
+            receipts_cache_lock.remove(outcome.execution_outcome.id.to_string().as_str());
```

Comment:
Are we sure whether the parent tx hash is no longer needed in the cache at this point?

Reply:
@khorolets It is a fair question from @telezhnaya. While we know it is a correct invariant, we should communicate this explicitly.

Reply:
@telezhnaya I'm pretty sure. Are you thinking about something I could miss?

@frol do you want me to extend the comment about it or what?

Reply:
I had no idea of the bad scenario. That was just unclear for me from the brief reading of the code, why can we drop this cache entry without thinking that it could be useful again.

After the second (again brief) reading, I got the invariant this way: we will appear here once for each receipt. So we need each line in the cache only once; that's why we can drop it after the usage.

In this case, I see another potential problem: our cache can grow infinitely. It's a rare case and I'm sure we will not catch this point, but it's better not to rely on that.
Maybe we want to add removing the item from cache after 100 blocks passed and noone came for it.

Reply:
@telezhnaya I've replaced `HashMap` with `cached::SizedCache` to address the 
> our cache can grow infinitely